src/fight.cpp

```diff
@@ -5522,21 +5547,9 @@ void roll_individual_initiative(struct char_data *ch)
       else
         GET_INIT_ROLL(ch) += 20;
     }
-    // Here we set the Surprise flag to our target if conditions are met. Clearing of flag and alert status are handled
-    // in hit_with_multi_weapon_toggle() as soon as a suprised NPC gets damaged in combat. Otherwise we can't
-    // properly enter the surprise state due to mobs being continously alert. Because we were handling this in
-    // init state and either the flag was getting stripped or the mob became alert before we ever tried to hit it.
-    // Also the no defense condition happens in the hit_with_multi_weapon_toggle() and we can't remove surprise
-    // flag until we process all that.
-    if (FIGHTING(ch)
-        && IS_NPC(FIGHTING(ch))
-        && !MOB_FLAGGED(FIGHTING(ch), MOB_INANIMATE)
-        && GET_MOBALERT(FIGHTING(ch)) == MALERT_CALM
-        && success_test(GET_REA(ch), 4) > success_test(GET_REA(FIGHTING(ch)), 4)) {
-      GET_INIT_ROLL(FIGHTING(ch)) = 0;
-      act("You surprise $n!", TRUE, FIGHTING(ch), 0, ch, TO_VICT);
```

Comment:
What was the decision process behind moving this code from here to the previous function?

Reply:
Setting init to 0 instead of calling roll_individual_initiative meant that it was possible for the opponent to change mobalert state while waiting for the fight to start, thus blocking surprise.